backend/src/main/java/de/jankahle/capstone/service/RecipeTextFilterService.java

```diff
@@ -0,0 +1,97 @@
+package de.jankahle.capstone.service;
+
+import de.jankahle.capstone.model.Ingredient;
+import de.jankahle.capstone.model.Recipe;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.regex.*;
+
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class RecipeTextFilterService {
+
+    public Recipe parseStringToRecipe(String recipeString) {
+        return Recipe.builder().build();
+    }
+
+    Optional<Ingredient> parseIngredient(String recipeLine) {
+
+        ArrayList<String> matcherResults = filterLineForIngredientDetails(recipeLine);
+
+        if (matcherResults.isEmpty())
+            return Optional.empty();
+
+        String matcherResult = normalizeMatcherResults(matcherResults);
+
+        String[] ingredientData = matcherResult.split(" ");
+
+        String name = constructIngredientName(ingredientData);
+        String amount = ingredientData[0];
+        String measurementUnit = ingredientData.length > 2 ? ingredientData[1] : "";
+
+        Ingredient parsedIngredient = Ingredient.builder()
+                .name(name)
+                .amount(amount)
+                .measurementUnit(measurementUnit)
+                .build();
+
+        return Optional.of(parsedIngredient);
+    }
+
+    /*
+        <Filter Description>
+        \\d+ we expect an listing of an ingredient starts with a number/the amount of the ingredient this can have multiple digits.
+        [\/.,]?\\d? the amount may be partial so we need to consider things like 1/2 or 1.5, this is optional.
```

Comment:
sometimes recipes will also include a minus to indicate a rough amount e.g. `1-2 Knoblauchzehen`

Reply:
Good point, I'll add that